### ISSUE TITLE
Still not possible to sync zec / zcash on Ledger Live

### ISSUE THREAD

**Issue by hapklaar** [LedgerHQ/ledger-live-desktop#1439]
On Windows 10 x64 it is still not possible for me and others to sync my ZCash wallet. The sync attempt always fails with a synchronization error. I reported this issue many weeks ago to Ledger support, but got no response other than that they are looking into it...

Tried up until version 1.1.7. Happy to also supply logs privately to devs.

**Comment by gre** [comment]
do you have a support ID with the logs attached? thanks

**Comment by hapklaar** [comment]
Sure. Id is 192247

Thanks
________________________________

**Comment by KhalilBellakrid** [comment]
Hello @hapklaar ,
I've already analysed your logs and it appears that you are mining, which is the issue here, your synchronisation takes a lot of time and from frontend side we have timeouts to contain synchros ! 

@gre we can probably avoid this timeouts by reseting them at each new call to API for one batch of addresses that we are trying to synchronise, when a batch of addresses is getting synced, API truncate response to 200 Txs (if I remember) but set a flag `truncated = true` so we advance in synchro on same batch of addresses but starting from last explored block in previous call, that's the moment we should reset our timeout and not at the very end of synchro of one batch of addresses (or worse an account), what do you think ? Is it already implemented like that ?

**Comment by hapklaar** [comment]
Hi @KhalilBellakrid,
Thank you for your reply. I do indeed transfer proceeds of mining to this wallet. I just counted and it has a total of 424 transactions. Is this too much for the app to handle? The old "Ledger Wallet Bitcoin" has no problem at all with it and syncs in a couple of seconds.

**Comment by kf6kjg** [comment]
With much rejoicing it works as of v1.1.9 with the removal of the coreSyncTimeout. Took a couple of tries, but it got sync'd.

**Comment by hapklaar** [comment]
Unfortunately not for me. It tries for much longer without this timeout, but eventually still times out with another timeout error. I've updated ticket 192247 with a new log.

**Comment by usk0411** [comment]
Unfortunately, I faced the same problem.
Does the problem still remain unsolved?

**Comment by meowssir** [comment]
hi fellows. firstly great work. now for the complainy part - so i mine Zcash and have accumulated quite a number of txns (613) to be precise. i am still facing timeouts even on version 1.6.0. I don't understand the technical details in this [comment](https://github.com/LedgerHQ/ledger-live-desktop/issues/1439#issuecomment-415416663) but would it not be an idea to have adjustable flags / parameters specific to miners to avoid timeouts? Also, is this issue only happening with Zcash? 

thanks and my support case is 278522 but no real update on that except to wait..

**Comment by gre** [comment]
We'll remove that timeout as soon as we parallelize the syncs. It should happen soon after #1888 is merged.
I'm not sure if it will fix your case but should help.
We also plan (Q2) to have a better visibility of which account is syncing like it's done in the mobile app.

**Comment by blueoceanmo** [comment]
I am mining to Ledger Zcash wallet and can not access my funds either.  Ledger Live continously says it is still synching.  I have tried exporting the private key via BIP39 decoder and that doesn't work.  I just want my funds out.  How can I fix this???

**Comment by gre** [comment]
@blueoceanmo you can try our latest 1.8.0 that should no longer have any timeout.

**Comment by chaxz93** [comment]
Hi I mine ZCash to the address controlled by my Ledger Nano... I have been unable to extract my funds for over a year! I am on the latest firmware (1.3.9) 
![Screenshot from 2019-05-22 08-56-09](https://user-images.githubusercontent.com/1481037/58145612-61c51c80-7c70-11e9-9fb9-39c111584aa2.png)
and Ledger Live versions! Please please help!

**Comment by hapklaar** [comment]
I don't think you will ever get access with Ledger Live to your coins. It has been like this for a long time and Ledger doesn't seem able to fix it, even though everything was fine when using the old Ledger Wallet.

I just took the private key I have for my Ledger and imported into ZecWallet. I was able to move all coins to a different address.

**Comment by chaxz93** [comment]
@hapklaar 

Could you help me replicate your exact steps to get my private key for my ledger and import into the ZecWallet? I recently tried to do it for Bitcoin but failed to figure out my private key. The steps I took were:

1) Navigate to https://iancoleman.io/bip39/ 
2) Typed in my 24-word Ledger Nano S master seed in the **BIP39 Mnemonic** field (lowercase chars)
3) Typed my 8 digit pin on the physical Ledger Nano S device in the **BIP39 Passphrase (optional)** field
4) Selected BTC - Bitcoin in the **coin** field (this time it should be ZEC - Zcash?) 
5) Leave the **BIP32 Root Key** field blank
6) Which derivation path should I chose ?
7) Enter 0 for the **Account** field
8) Enter 0 for the **Internal/External** field?

Would be indebted for your help.
Cheers,

**Comment by chaxz93** [comment]
> I don't think you will ever get access with Ledger Live to your ZCash coins. It has been like this for a long time and Ledger doesn't seem able to fix it, even though everything was fine when using the old Ledger Wallet.
> 
> I just took the private key I have for my Ledger and imported into ZecWallet. I was able to move all coins to a different address.

@hapklaar 

Could you help me replicate your exact steps to get my private key for my ledger and import into the ZecWallet? I recently tried to do it for Bitcoin but failed to figure out my private key. The steps I took were:

1) Navigate to https://iancoleman.io/bip39/ 
2) Typed in my 24-word Ledger Nano S master seed in the **BIP39 Mnemonic** field (lowercase chars)
3) Typed my 8 digit pin on the physical Ledger Nano S device in the **BIP39 Passphrase (optional)** field
4) Selected BTC - Bitcoin in the **coin** field (this time it should be ZEC - Zcash?) 
5) Leave the **BIP32 Root Key** field blank
6) Which derivation path should I chose ?
7) Enter 0 for the **Account** field
8) Enter 0 for the **Internal/External** field?

Would be indebted for your help.
Cheers,

**Comment by chaxz93** [comment]
Thanks figured it out and sold ZEC! However BitOasis unfortunately delisted it without any explanation along with Monero

**Comment by hi202020ch** [comment]
> I don't think you will ever get access with Ledger Live to your ZCash coins. It has been like this for a long time and Ledger doesn't seem able to fix it, even though everything was fine when using the old Ledger Wallet.
> 
> I just took the private key I have for my Ledger and imported into ZecWallet. I was able to move all coins to a different address.

May I know how to get private key from Ledger Wallet?

**Comment by chaxz93** [comment]
> > I don't think you will ever get access with Ledger Live to your ZCash coins. It has been like this for a long time and Ledger doesn't seem able to fix it, even though everything was fine when using the old Ledger Wallet.
> > I just took the private key I have for my Ledger and imported into ZecWallet. I was able to move all coins to a different address.
> 
> May I know how to get private key from Ledger Wallet?

Hey its very easy. Just go to this website:

1. https://iancoleman.io/bip39/
2. Put in your master seed words with spaces in the **BIP39 Mnemonic** field.
3. Leave the **BIP39 Passphrase (optional)** field empty! It is not the 8 digit pincode you physically punch into the Ledger Nano.
4. For the **Coin** field, pick _ZEC -Zcash_ from the dropdown.
5. Scroll down the page to the **Derivation Path** part of the page. There will be a whole list of Addresses and Private Key Pairs. Find the Address of your Ledger Nano ZCash Wallet that your funds have been stuck on / synchronizing forever, and find the corresponding private key pair.
6. Download Coinomi's wallet on your Windows/Linux/Mac from here:
https://www.coinomi.com/downloads/
7. Instead of creating a new wallet, import using the private you got from step 5. Be sure to delete any local downloads/files on your laptop in which you write/copy/paste the retrieved private key in step 5.

Good Luck! If you have any further questions, feel free to email me at srinjoy@dragonchain.com

**Comment by gre** [comment]
I would really recommend NOT putting your seed in any software, ever. Seed should only be used in hardware wallet so it's not exposed on the software side. 
If you use your seed, consider all the funds associated to it compromised and use a new one.
